File: programacao/python/respostas/selecao/30.py
aulas = int(input("Total de aulas: "))
faltas = int(input("Total de faltas: "))
p1 = float(input("Nota da prova 1: "))
p2 = float(input("Nota da prova 2: "))

frequencia = (aulas - faltas) / aulas * 100.0
media = (p1 + p2) / 2.0

if frequencia < 75.0:
    print("Reprovado por frequencia")
elif media >= 6.0:
    print(f"Aprovado com media {media:.1f}")
elif media < 4.0:
    print(f"Reprovado com media {media:.1f}")
else:
    final = float(input("Nota da prova final: "))
    if final >= 6.0:
        print(f"Aprovado na prova final com media {final:.1f}")
    else:
        media_final = final if final > media else media
        print(f"Reprovado na prova final com media {media_final:.1f}")

File: programacao/c/respostas/selecao/30.c
#include <stdio.h>

int main(void) {
    int aulas, faltas;
    float p1, p2, media, frequencia, final;

    printf("Total de aulas: ");
    scanf("%d", &aulas);
    printf("Total de faltas: ");
    scanf("%d", &faltas);
    printf("Nota da prova 1: ");
    scanf("%f", &p1);
    printf("Nota da prova 2: ");
    scanf("%f", &p2);

    frequencia = (float)(aulas - faltas) / aulas * 100.0;
    media = (p1 + p2) / 2.0;

    if (frequencia < 75.0) {
        printf("Reprovado por frequencia\n");
    } else if (media >= 6.0) {
        printf("Aprovado com media %.1f\n", media);
    } else if (media < 4.0) {
        printf("Reprovado com media %.1f\n", media);
    } else {
        printf("Nota da prova final: ");
        scanf("%f", &final);
        if (final >= 6.0) {
            printf("Aprovado na prova final com media %.1f\n", final);
        } else {
            printf("Reprovado na prova final com media %.1f\n", (final > media ? final : media));
        }
    }

    return 0;
}
